tools/chainsync-loadtest/src/concurrency/rate_limiter.rs

```diff
@@ -0,0 +1,54 @@
+use crate::concurrency::Ctx;
+use std::sync::Arc;
+use tokio::time;
+
+struct RateLimiter_ {
+    interval: time::Duration,
+    burst: u64,
+    tokens: u64,
+    ticks_processed: u64,
+    start: time::Instant,
+}
+
+impl RateLimiter_ {
+    fn ticks(&self, t: time::Instant) -> u64 {
+        return ((t - self.start).as_secs_f64() / self.interval.as_secs_f64()) as u64;
```

Comment:
This can sometimes (in very rare instances) produce a `(0.0/0.0) as u64 = 0` for `self.interval = 0ns`, even though the intent is probably for it to be a `+INF as u64` at all times.

Reply:
The intent is to panic if interval is 0.